channeldb/codec.go

```diff
@@ -213,6 +213,9 @@ func WriteElement(w io.Writer, element interface{}) error {
 			}
 		}
 
+	case lnwire.DeliveryAddress:
```

Comment:
I don't think we need to add this here? we can just pass the bytes directly from the put method.

Reply:
We don't need this for put, but do need it to read into a `DeliveryAddress` type on the channel struct.
Is it ok to have `ReadElement` but not `WriteElement` for a type?

Can also get around if for read by reading into a `[]byte` then casting to `DeliveryAddress` but that feels messier?

I'm going to remove for write for now, and can do this ^ workaround for read if it's necessary to have `ReadElement`/`WriteElement` implemented in pairs.

Reply:
Yeah, we should definitely have pairs. I was thinking just doing the casting.

Reply:
Any specific reason we don't want to add delivery address to `Read/WriteElement`?
Casting works fine for writes, but it's not as smooth for reads because our existing pattern is to pass in a pointer to the destination. 

Could have something like `getOptionalUpfrontShutdownScript` return the value (rather than setting the pointer's value in the function) but I don't think that's better than just having delivery addr is codec?

Reply:
Just to avoid cluttering the codec with types, as it is really just an alias for `[]byte`